internal/services/storage/storage_account_customer_managed_key_resource.go

```diff
@@ -166,13 +189,14 @@ func resourceStorageAccountCustomerManagedKeyCreateUpdate(d *pluginsdk.ResourceD
 					},
 				},
 				EncryptionIdentity: &storage.EncryptionIdentity{
-					EncryptionUserAssignedIdentity: utils.String(userAssignedIdentity),
+					EncryptionUserAssignedIdentity:      utils.String(userAssignedIdentity),
+					EncryptionFederatedIdentityClientID: utils.String(federatedIdentityClientID),
```

Comment:
Probably we shall optioanlly set this only when `federated_identity_client_id` is specified in the config?